server/src/db/catalog.rs

```diff
@@ -117,6 +133,72 @@ impl Catalog {
             .cloned()
             .context(UnknownPartition { partition_key })
     }
+
+    /// Return the chunks for specific partition
+    pub fn chunks(&self, partition_key: &str) -> Vec<Arc<DBChunk>> {
+        let partition = match self.partition(partition_key) {
+            Some(partition) => partition,
+            None => return vec![],
+        };
+
+        let partition = partition.read();
+
+        partition
+            .chunks()
+            .map(|chunk| {
+                let chunk = chunk.read();
+                DBChunk::snapshot(&chunk)
+            })
+            .collect()
+    }
+}
+
+impl SchemaProvider for Catalog {
+    fn as_any(&self) -> &dyn Any {
+        self as &dyn Any
+    }
+
+    fn table_names(&self) -> Vec<String> {
+        let mut names = StringSet::new();
+
+        self.partitions().for_each(|partition| {
+            let partition = partition.read();
+            partition.chunks().for_each(|chunk| {
+                let chunk = chunk.read();
+                let db_chunk = DBChunk::snapshot(&chunk);
+                db_chunk.all_table_names(&mut names);
+            })
+        });
+
+        names.into_iter().collect::<Vec<_>>()
```

Comment:
I know you're moving this code, not adding this as new, but while we're changing it... this turbofish isn't necessary with the function's return type so this can be:

```suggestion
        names.into_iter().collect()
```